src/dmd/parse.d

```diff
@@ -3497,6 +3498,25 @@ final class Parser(AST) : Lexer
             t = parseVector();
             break;
 
+        case TOK.traits:
+        {
+            AST.Expression e = parsePrimaryExp();
+            AST.TraitsExp te = cast(AST.TraitsExp) e;
+            if (!te)
+            {
+                error("invalid `__traits` expression");
+                break;
+            }
+            else if (strcmp(te.ident.toChars, "getMember") != 0)
+            {
+                error("invalid `__traits`, only `getMember` can be aliased and not `%s`",
```

Comment:
Should this really be handled by the parser?

Reply:
No. It's a semantic thing that's handled eagerly.